test-utils/src/dev/db.rs

```diff
@@ -1542,7 +1542,7 @@ mod test {
 
         // unexpected contents in initial listen URL (extra param)
         let error = make_pg_config(
-            "postgresql://root@127.0.0.1:45913/foobar?application_name=foo",
+            "postgresql://root@[::1]:45913/foobar?application_name=foo",
```

Comment:
I confirmed that these IP changes do not affect my ability to use `omicron-dev run-all` — why did they change to IPv6, though?

Reply:
- Basically everything on the underlay network runs on IPv6
- Localhost, decidedly, it "not really" part of the underlay network, but just a thing we use under test
- However, a good portion of the wrappers around internal DNS access **assume** we're using IPv6. This is not a hard blocker, it just happens to be a common call where we expect underlay addresses: https://github.com/oxidecomputer/omicron/blob/d70f7ff0c75bdeb9651ebcb4c3a71250c07c75a3/internal-dns/src/resolver.rs#L126
- Switching to using IPv6 under tests means we can use the same DNS calls without modification

The alternative would be:
- Keep the test code IPv4
- Modify the non-test code to cope with both IPv4 and IPv6 addresses on the underlay

Reply:
FWIW I **can** go do the alternative, but I figured "make test and prod more aligned on addressing scheme" seemed like a worthwhile goal

Reply:
Yeah, sounds good